src/Main.elm

```diff
@@ -55,8 +62,12 @@ init flags url key =
       , habitLog = habitLog
       , editing = False
       , now = Time.millisToPosix 0
+      , viewportSize = ViewportSize 0
```

Comment:
Instead of initializing to '0', i wonder if we could have a better type for the model? something like:
```
type Model = Booting | MainApp Context

type alias Context = {
  habit: Habit
  , ...
```

Reply:
I see yes this is much more Elm-y!